app/src/main/java/org/wikipedia/page/PageTitle.kt

```diff
@@ -0,0 +1,212 @@
+
+    /**
+     * Notes on the `namespace` field:
+     * The localised namespace of the page as a string, or null if the page is in mainspace.
+     *
+     * This field contains the prefix of the page's title, as opposed to the namespace ID used by
+     * MediaWiki. Therefore, mainspace pages always have a null namespace, as they have no prefix,
+     * and the namespace of a page will depend on the language of the wiki the user is currently
+     * looking at.
+     *
+     * Examples:
+     * * [[Manchester]] on enwiki will have a namespace of null
+     * * [[Deutschland]] on dewiki will have a namespace of null
+     * * [[User:Deskana]] on enwiki will have a namespace of "User"
+     * * [[Utilisateur:Deskana]] on frwiki will have a namespace of "Utilisateur", even if you got
+     * to the page by going to [[User:Deskana]] and having MediaWiki automatically redirect you.
+     */
+
+    constructor(namespace: String?, text: String, fragment: String?, thumbUrl: String?, wiki: WikiSite) :
+            this(namespace, wiki, text, fragment, thumbUrl, null, null, null)
+
+    constructor(text: String?, wiki: WikiSite, thumbUrl: String?, description: String?, displayText: String?) :
+            this(text, wiki, thumbUrl) {
+        this._displayText = displayText
+        this.description = description
+    }
+
+    constructor(
+        text: String?,
+        wiki: WikiSite,
+        thumbUrl: String?,
+        description: String?,
+        displayText: String?,
+        extract: String?
+    ) : this(text, wiki, thumbUrl, description, displayText) {
+        this.extract = extract
+    }
+
+    constructor(namespace: String?, text: String, wiki: WikiSite) :
+            this(namespace, text, null, null, wiki)
+
+    @JvmOverloads
+    constructor(title: String?, wiki: WikiSite, thumbUrl: String? = null) :
+            this(null, wiki, title.orEmpty(), null, thumbUrl, null, null, null) {
+        // FIXME: Does not handle mainspace articles with a colon in the title well at all
+        var text = title.orEmpty().ifEmpty { getMainPageForLang(wiki.languageCode) }
+
+        // Split off any fragment (#...) from the title
+        var parts = text.split("#".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
+        text = if (parts.isNotEmpty()) parts[0] else ""
+        fragment = if (parts.size > 1) {
+            addUnderscores(decodeURL(parts[1]))
+        } else {
+            null
+        }
+
+        // Remove any URL parameters (?...) from the title
+        parts = text.split("\\?".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
+        if (parts.size > 1 && parts[1].contains("=")) {
+            text = parts[0]
+        }
+        parts = text.split(":".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
+        if (parts.size > 1) {
+            val namespaceOrLanguage = parts[0]
+            if (listOf(*Locale.getISOLanguages()).contains(namespaceOrLanguage)) {
```

Comment:
Would it be possible to just do `Locale.getISOLanguages().contains(namespaceOrLanguage)`?

Reply:
yes!